File: Exercise/Medium/13. Print Number in Large Format/Code/Print Number in Large Format.py
def print_large_digit(digit):
    if digit == 0:
        print(" ##### ")
        print("#     #")
        print("#     #")
        print("#     #")
        print("#     #")
        print("#     #")
        print(" ##### ")
    elif digit == 1:
        print("   #   ")
        print("  ##   ")
        print("   #   ")
        print("   #   ")
        print("   #   ")
        print("   #   ")
        print(" ##### ")
    elif digit == 2:
        print(" ##### ")
        print("#     #")
        print("      #")
        print(" ##### ")
        print("#      ")
        print("#      ")
        print("#######")
    elif digit == 3:
        print(" ##### ")
        print("#     #")
        print("      #")
        print(" ##### ")
        print("      #")
        print("#     #")
        print(" ##### ")
    elif digit == 4:
        print("#      ")
        print("#    # ")
        print("#    # ")
        print("#######")
        print("     # ")
        print("     # ")
        print("     # ")
    elif digit == 5:
        print("#######")
        print("#      ")
        print("#      ")
        print(" ##### ")
        print("      #")
        print("#     #")
        print(" ##### ")
    elif digit == 6:
        print(" ##### ")
        print("#     #")
        print("#      ")
        print("###### ")
        print("#     #")
        print("#     #")
        print(" ##### ")
    elif digit == 7:
        print("#######")
        print("     # ")
        print("    #  ")
        print("   #   ")
        print("  #    ")
        print(" #     ")
        print("#      ")
    elif digit == 8:
        print(" ##### ")
        print("#     #")
        print("#     #")
        print(" ##### ")
        print("#     #")
        print("#     #")
        print(" ##### ")
    elif digit == 9:
        print(" ##### ")
        print("#     #")
        print("#     #")
        print("###### ")
        print("      #")
        print("#     #")
        print(" ##### ")

def print_large_number(number):
    if number == 0:
        print_large_digit(0)
        return

    digits = []
    while number > 0:
        digits.append(number % 10)
        number //= 10

    for digit in reversed(digits):
        print_large_digit(digit)

if __name__ == "__main__":
    number = int(input())
    print_large_number(number)

File: Exercise/Medium/13. Print Number in Large Format/Code/Print Number in Large Format.java
import java.util.Scanner;

public class PrintLargeNumber {
    public static void printLargeDigit(int digit) {
        switch (digit) {
            case 0:
                System.out.println(" ##### ");
                System.out.println("#     #");
                System.out.println("#     #");
                System.out.println("#     #");
                System.out.println("#     #");
                System.out.println("#     #");
                System.out.println(" ##### ");
                break;
            case 1:
                System.out.println("   #   ");
                System.out.println("  ##   ");
                System.out.println("   #   ");
                System.out.println("   #   ");
                System.out.println("   #   ");
                System.out.println("   #   ");
                System.out.println(" ##### ");
                break;
            case 2:
                System.out.println(" ##### ");
                System.out.println("#     #");
                System.out.println("      #");
                System.out.println(" ##### ");
                System.out.println("#      ");
                System.out.println("#      ");
                System.out.println("#######");
                break;
            case 3:
                System.out.println(" ##### ");
                System.out.println("#     #");
                System.out.println("      #");
                System.out.println(" ##### ");
                System.out.println("      #");
                System.out.println("#     #");
                System.out.println(" ##### ");
                break;
            case 4:
                System.out.println("#      ");
                System.out.println("#    # ");
                System.out.println("#    # ");
                System.out.println("#######");
                System.out.println("     # ");
                System.out.println("     # ");
                System.out.println("     # ");
                break;
            case 5:
                System.out.println("#######");
                System.out.println("#      ");
                System.out.println("#      ");
                System.out.println(" ##### ");
                System.out.println("      #");
                System.out.println("#     #");
                System.out.println(" ##### ");
                break;
            case 6:
                System.out.println(" ##### ");
                System.out.println("#     #");
                System.out.println("#      ");
                System.out.println("###### ");
                System.out.println("#     #");
                System.out.println("#     #");
                System.out.println(" ##### ");
                break;
            case 7:
                System.out.println("#######");
                System.out.println("     # ");
                System.out.println("    #  ");
                System.out.println("   #   ");
                System.out.println("  #    ");
                System.out.println(" #     ");
                System.out.println("#      ");
                break;
            case 8:
                System.out.println(" ##### ");
                System.out.println("#     #");
                System.out.println("#     #");
                System.out.println(" ##### ");
                System.out.println("#     #");
                System.out.println("#     #");
                System.out.println(" ##### ");
                break;
            case 9:
                System.out.println(" ##### ");
                System.out.println("#     #");
                System.out.println("#     #");
                System.out.println("###### ");
                System.out.println("      #");
                System.out.println("#     #");
                System.out.println(" ##### ");
                break;
        }
    }

    public static void printLargeNumber(int number) {
        if (number == 0) {
            printLargeDigit(0);
            return;
        }

        int[] digits = new int[10];
        int count = 0;

        while (number > 0) {
            digits[count++] = number % 10;
            number /= 10;
        }

        for (int i = count - 1; i >= 0; --i) {
            printLargeDigit(digits[i]);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int number = scanner.nextInt();

        printLargeNumber(number);

        scanner.close();
    }
}




